GitExtUtils/GitCommandConfiguration.cs

```diff
@@ -27,6 +27,7 @@ static GitCommandConfiguration()
             Default.Add(new GitConfigItem("diff.mnemonicprefix", "false"), "diff");
             Default.Add(new GitConfigItem("diff.ignoreSubmodules", "none"), "diff", "status");
             Default.Add(new GitConfigItem("core.safecrlf", "false"), "diff");
+            Default.Add(new GitConfigItem("protocol.file.allow", "always"), "submodule"); // git version 2.38.1 and later disabled file protocol by default
```

Comment:
Should this be a popup for the user to confirm?
(Tests obviously needs this.)

Reply:
I agree with @gerhardol, since this may be a security-related config, we shouldn't sneakily enable it. We should ask for an explicit permission from a user. Also, I think, it's a good idea to expose the setting under Git > Config.

Reply:
...also it's good to check how well the app fares if the setting is set to `never`.

Reply:
Without this change user is the default.
```git submodule add` and ```git submodule update --init --recursive``` fail for file protocol repo paths.  Wjat this does is break --recursive if not always is set.  Always is an opt in to say yes I mean it.  The user is initiating the command.  So it is ok.

Reply:
https://www.git-scm.com/docs/git#Documentation/git.txt-codeGITPROTOCOLFROMUSERcode

Reply:
So tests need this so I can move to the test config setup and I can work another branch to do a some kind of ui that says something like...

"...." command about to be run.  Do you trust it.  and if yes...add that to the options of that command.  I'll add a needs userTrust feature in a different pull request.  I'll change this pull request for just tests.

Reply:
For sure change in tests.
I can agree that setting by default in commands is good for users, most will not understand the reason anyway and just find the popup annoying.
To follow that "GE is a GUI for Git" with no major deviation from default behavior, there should be a popup.